test/03_device-state.ts

```diff
@@ -510,6 +555,316 @@ export default () => {
 									api_heartbeat_state: DeviceOnlineStates.Offline,
 								});
 							});
+
+							describe('when the write cache expires while the device is on Timeout (eg API stops serving requests)', function () {
+								let device3: fakeDevice.Device;
+								const device3ChangeEventSpy = sinon.spy();
+
+								before(async function () {
+									device3 = await fakeDevice.provisionDevice(
```

Comment:
```suggestion
									// Provision a device and wait for its heartbeat to be Online
									device3 = await fakeDevice.provisionDevice(
```

Reply:
That should be clear from the describe "when the write cache expires while the device is on Timeout" but fine.